src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -113,7 +113,17 @@ private MethodIL CreateMethodIL(MethodDesc method)
                         return result;
                 }
 
-                return EcmaMethodIL.Create((EcmaMethod)method);
+                MethodIL methodIL = EcmaMethodIL.Create((EcmaMethod)method);
+                if (methodIL != null)
+                    return methodIL;
+                
+                if (!((EcmaMethod)method).ImplAttributes.HasFlag(System.Reflection.MethodImplAttributes.InternalCall) && 
```

Comment:
I think it might be worth adding `IsInternalCall` to `MethodDesc` in MethodDesc.CodeGen.cs.